src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -236,30 +277,30 @@ class SignUpModalController {
     }
   }
 
-  async loadDonorDetails () {
+  loadDonorDetails () {
     this.loadingDonorDetails = true
-    return new Promise((resolve) => {
-      this.orderService.getDonorDetails().subscribe(
-        (data) => {
-          this.loadingDonorDetails = false
-
+    return new Observable(observer => {
+      this.orderService.getDonorDetails().subscribe({
```

Comment:
Instead of making a new observable, I think you can use `map` to transform the donor details and `do` to inject error handling:

```js
return this.orderService.getDonorDetails().map((data) => {
 // Update donorData as needed
 return donorData;
}.do({
  error: () => {
    // Do error handling
  },
  finally: () => {
    this.loadingDonorDetails = false
  }
})
```